Downstream RPD as designed by gusxyz

ilovehans10 opened the pull request:
## About the PR
This PR impliments the RPD as found in [upstream pr #32675](https://github.com/space-wizards/space-station-14/pull/32675).
NOTE: I had no hand in developing this, and am only porting it from upstream, as its maintainer discussion has been long-winded.
This is my first time contributing, so please provide any suggestions.
~~I also know that on other servers that have implemented this in the implementation PR they also add it to atmos lockers, so if that is desired I can add that to this PR too. I am going to mark this PR as a draft until I receive a verdict on whether this should be added to Atmos lockers.~~ I have added this to both Atmos and CE lockers at the moment, and if maintainers want me to just put it in one or the other or neither, just let me know and I'll change it. This way I can also make it open instead of draft.

## Why / Balance
Allows for rapid building of atmos devices at the cost of compressed matter.

## Technical details
The Rapid Piping Device is a variation of the Rapid Construction Device that is specifically designed for atmospherics work. More information can be found on the original pull request.

## Media

![Screenshot From 2025-01-18 10-05-59](https://github.com/user-attachments/assets/3860563c-6f28-431e-a8f3-906e12ce321a)

![Screenshot From 2025-01-18 10-06-53](https://github.com/user-attachments/assets/0b9cd55c-378e-461d-a1c5-0ebd406e9c92)

For more see original pull request

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**
:cl:
- add: Rapid Piping Device

Ermucat commented:
Should the files be under _Harmony namespace? Or is it a early implementation of [#32675](https://github.com/space-wizards/space-station-14/pull/32675)

ilovehans10 commented:
> Should the files be under _Harmony namespace? Or is it a early implementation of https://github.com/space-wizards/space-station-14/pull/32675

> This PR impliments the RPD as found in [upstream pr #32675.](https://github.com/space-wizards/space-station-14/pull/32675).

I used the same commits as found upstream, so that shouldn't cause any merge conflicts. The only thing that I added to the PR was adding the RPD to atmos and CE lockers, which I am unsure how I would include in a separate namespace. I tried to add comments for the three lines I changed, indicating that they were harmony specific changes.

Ermucat commented:
You commented the lockers correctly. (I think)

DieselMohawk commented:
This lacks any and all Harmony notation, it'll make it harder for us in the future, please add notation in the box notation format

# Start Harmony Change: RPD -- Done by gusyzx
Added content
# End Harmony Change

DieselMohawk commented:
The notation is for if the upstream PR gets merged so we can revert this PR and make way for it.

FluffMe commented:
> The notation is for if the upstream PR gets merged so we can revert this PR and make way for it.

It's literal upstream commits. So it's fine

DieselMohawk reviewed (approved):
Sprites look good.

FluffMe reviewed (approved):
Looks good, but DNM for a week or so.

Please keep an eye out on upstream PR in case of additional cherry-picks or a merge.

Also slapping admin input in case they want to change which lockers have the RPD. Currently it's atmos and CE.

FluffMe commented:
Bugged. Crashes client. Needs reverting

Zap527 commented:
It doesn't crash the client but it makes it unusable and you need to straight up close it to be able to do anything with UIs.

FluffMe commented:
Since autorevert doesn't work, one (or more) of the files might have been changed in recent upstream merge and introduced the issue
